### PR TITLE
Delete server command

### LINKED ISSUE #11: Delete server command
```
spinupwp servers:delete <server_id> --force
```

As with the SSH commands, when no `<server_id>` is supplied we should prompt them for the server by listing them.

This should show a warning for the user to confirm the deletion. When the `--force` flag is supplied, no warning will be shown.

### PR DESCRIPTION
**Pull request by danielmlozano**
Resolves #11 

Includes the `--force` and `--delete-on-provider` flags.

### CODE REVIEW

File: app/Commands/Servers/DeleteCommand.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    public function action(): int
+    {
+        $serverId = $this->argument('id');
+
+        if (empty($serverId)) {
+            $serverId = $this->askToSelectServer('Which server would you like to delete?');
+        }
+
+        $delete = $this->option('force') || $this->confirm('Are you sure you want to delete this server?');
```
Comment (A5hleyRich) [inline app/Commands/Servers/DeleteCommand.php:23]: Before asking them to confirm, let's show an alert:

> You're about to delete "SERVER NAME"

Then update the confirm wording to:

> Do you wish to continue?

File: app/Commands/Servers/DeleteCommand.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Commands\Servers;
+
+use App\Commands\BaseCommand;
+
+class DeleteCommand extends BaseCommand
+{
+    protected $signature = 'servers:delete {id?} {--force} {--profile=} {--delete-on-provider}';
```
Comment (A5hleyRich) [inline app/Commands/Servers/DeleteCommand.php:9]: Let's tidy this up and provide descriptions:

```
{server_id? : The server to delete}
{--d|delete-on-provider : Delete the server from the server provider (DigitalOcean, etc.)}
{--f|force : Delete the server without prompting for confirmation}
{--profile=}
```

**Review by A5hleyRich — CHANGES_REQUESTED** [review]
@danielmlozano does this work against a local Spinup install for you? I get the following error:

> Undefined index: event_id

**Comment by danielmlozano** [comment]
Took me a while finding my own error. Since I cannot hit the API URL using Valet SSL (it throws a `curl` error), I added the "`verify => false`" option in the custom client, and that was causing the `event_id` error.

File: app/Commands/Servers/DeleteCommand.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        if (empty($serverId)) {
+            $serverId = $this->askToSelectServer('Which server would you like to delete?');
+        }
+
+        $server = $this->spinupwp->servers->get($serverId);
+
+        $delete = (bool) $this->option('force');
+
+        if (!$delete) {
+            $this->warn("You're about to delete \"{$server->name}\"");
```
Comment (A5hleyRich) [inline app/Commands/Servers/DeleteCommand.php:32]: This should be an `alert`.

**Review by A5hleyRich — CHANGES_REQUESTED** [review]

File: app/Commands/BaseCommand.php [hunk 4]
```diff
@@ -39,17 +41,19 @@ public function handle(): int
...
-                    'base_uri'    => $this->config->get('api_url', $this->profile(), 'https://api.spinupwp.app/v1/'),
-                    'http_errors' => false,
-                    'headers'     => [
-                        'Authorization' => "Bearer {$this->apiToken()}",
-                        'Accept'        => 'application/json',
-                        'Content-Type'  => 'application/json',
-                        'User-Agent'    => 'SpinupWP/' . config('app.version'),
-                    ],
-                ]));
+            if (!$this->spinupwp->hasApiKey() || !empty($this->config->get('api_url', $this->profile()))) {
```
[the remark this answers is not in the document]
Comment (danielmlozano) [reply app/Commands/BaseCommand.php:44]: My bad, it was part of my debugging for the `event_id` issue.

**Comment by A5hleyRich** [comment]
💯

**Pull request merged**